examples/UNet/massachussets_road_segm.py

```diff
@@ -0,0 +1,340 @@
+    test_target_str = "https://www.cs.toronto.edu/~vmnih/data/mass_roads/test/map/"
+
+    f = open("mass_roads_train_data_download.sh", 'w')
+    g = open("mass_roads_train_target_download.sh", 'w')
+    for img_name in train_data_url:
+        f.write("wget -O data/training/sat_img/%sf "%img_name + train_data_str + img_name + "f" + "\n")
+        g.write("wget -O data/training/map/%s "%img_name + train_target_str + img_name + "\n")
+    f.close()
+    g.close()
+
+    f = open("mass_roads_validation_data_download.sh", 'w')
+    g = open("mass_roads_validation_target_download.sh", 'w')
+    for img_name in valid_data_url:
+        f.write("wget -O data/validation/sat_img/%s "%img_name + valid_data_str + img_name + "\n")
+        g.write("wget -O data/validation/map/%s "%img_name[:-1] + valid_target_str + img_name[:-1] + "\n")
+    f.close()
+    g.close()
+
+    f = open("mass_roads_test_data_download.sh", 'w')
+    g = open("mass_roads_test_target_download.sh", 'w')
+    for img_name in test_data_url:
+        f.write("wget -O data/test/sat_img/%s "%img_name + test_data_str + img_name + "\n")
+        g.write("wget -O data/test/map/%s "%img_name[:-1] + test_target_str + img_name[:-1] + "\n")
+    f.close()
+    g.close()
+
+def download_dataset():
+    os.system("sh mass_roads_train_data_download.sh &")
+    os.system("sh mass_roads_train_target_download.sh &")
+    os.system("sh mass_roads_validation_data_download.sh &")
+    os.system("sh mass_roads_validation_target_download.sh &")
+    os.system("sh mass_roads_test_data_download.sh &")
+    os.system("sh mass_roads_test_target_download.sh &")
+
+def load_data(folder):
+    images_sat = [img for img in os.listdir(os.path.join(folder, "sat_img")) if fnmatch.fnmatch(img, "*.tif*")]
+    images_map = [img for img in os.listdir(os.path.join(folder, "map")) if fnmatch.fnmatch(img, "*.tif*")]
+    assert(len(images_sat) == len(images_map))
+    images_sat.sort()
+    images_map.sort()
+    # images are 1500 by 1500 pixels each
+    data = np.zeros((len(images_sat), 3, 1500, 1500), dtype=np.uint8)
+    target = np.zeros((len(images_sat), 1, 1500, 1500), dtype=np.uint8)
+    ctr = 0
+    for sat_im, map_im in zip(images_sat, images_map):
+        data[ctr] = plt.imread(os.path.join(folder, "sat_img", sat_im)).transpose((2, 0, 1))
+        # target has values 0 and 255. make that 0 and 1
+        target[ctr, 0] = plt.imread(os.path.join(folder, "map", map_im))/255
+        ctr += 1
+    return data, target
+
+def batch_generator(data, target, BATCH_SIZE):
+    '''
+    just a simple batch iterator, no cropping, no rotation, no anything
+    '''
+    np.random.seed()
+    idx = np.arange(data.shape[0])
+    while True:
+        ids = np.random.choice(idx, BATCH_SIZE)
+        yield np.array(data[ids]), np.array(target[ids])
+
+def random_crop_generator(generator, crop_size=(128, 128)):
+    '''
+    yields a random crop of size crop_size
+    '''
+    np.random.seed()
```

Comment:
Both calls are unnecessary here, that is true. In this specific example I would agree that the calls could be removed. I would still like to keep the call in batch_generator because some people (like me) like to use multi threaded batch generation (which is necessary when applying for example elastic deformations). In absence of a np.random.seed(None) call each worker would generate exactly the same training data